generators/app/templates/src/main/java/package/service/_ElasticsearchIndexService.java

```diff
@@ -141,11 +147,30 @@ public void reindexAll() {
         }
         elasticsearchTemplate.putMapping(entityClass);
         if (jpaRepository.count() > 0) {
-            try {
-                Method m = jpaRepository.getClass().getMethod("findAllWithEagerRelationships");
-                elasticsearchRepository.save((List<T>) m.invoke(jpaRepository));
-            } catch (Exception e) {
-                elasticsearchRepository.save(jpaRepository.findAll());
+            // set up a list of relationships that should be eagerly loaded.
+            // if a JHipster entity field is a Set, it is a relationship
+            List<Method> relationshipGetters = Arrays.stream(entityClass.getDeclaredMethods())
+                .filter(method -> method.getReturnType().equals(Set.class))
+                .collect(Collectors.toList());
+
+            int size = 100;
+            for (int i = 0; i <= jpaRepository.count() / size; i++) {
+                Pageable page = new PageRequest(i, size);
+                log.info("Indexing page : " + i + " , nr " + size);
+                Page<T> results = jpaRepository.findAll(page);
+                results.map(result -> {
+                    // if there are any relationships to load, do it now
+                    relationshipGetters.forEach(method -> {
+                        try {
+                            // eagerly load the relationship set
+                            ((Set) method.invoke(result)).size();
```

Comment:
[An example of what this would normally look like](https://github.com/jhipster/generator-jhipster/blob/v2.x_maintenance/app/templates/src/main/java/package/service/_UserService.java#L185-L187)